examples/lang/erc721/src/lib.rs

```diff
@@ -0,0 +1,422 @@
+// Copyright 2018-2019 Parity Technologies (UK) Ltd.
+// This file is part of ink!.
+//
+// ink! is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// ink! is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with ink!.  If not, see <http://www.gnu.org/licenses/>.
+
+#![cfg_attr(not(feature = "std"), no_std)]
+
+use ink_core::{
+    env::DefaultSrmlTypes,
+    memory::format,
+    storage,
+};
+use ink_lang::contract;
+use ink_model;
+
+pub use core::result::Result;
+pub type EnvHandler = ink_model::EnvHandler<ink_core::env::ContractEnv<DefaultSrmlTypes>>;
+pub type TokenId = u32;
+pub type Counter = u32;
+pub type ErrNo = u32;
+
+const NOT_OWNER: ErrNo = 1;
+const INSERT_ERROR: ErrNo = 2;
+const TOKEN_EXISTS: ErrNo = 3;
+const NOT_APPROVED: ErrNo = 4;
+const TOKEN_NOT_FOUND: ErrNo = 5;
+const CAN_NOT_GET_COUNTER: ErrNo = 6;
+const ACCOUNT_ZERO_NOT_ALLOWED: ErrNo = 7;
+const CAN_NOT_REMOVE: ErrNo = 8;
```

Comment:
Please use a Rust `enum` for error kinds. You can implement a `to_u32` method on it if you want to use raw `u32` values instead upon returning to SRML contracts.